src/core/qgsvectorfilewriter.h

```diff
@@ -199,21 +199,24 @@ class CORE_EXPORT QgsVectorFileWriter
         virtual QVariant convert( int fieldIdxInLayer, const QVariant& value );
     };
 
-    /** Combination of CanAddNewLayer, CanAppendToExistingLayer, CanAddNewFieldsToExistingLayer or CanDeleteLayer
-      * @note Added in QGIS 3.0 */
-    typedef int EditionCapabilities;
+    enum EditionCapability
+    {
+      /** Flag to indicate that a new layer can be added to the dataset */
+      CanAddNewLayer                 = 1 << 0,
 
-    /** Flag to indicate that a new layer can be added to the dataset */
-    static const EditionCapabilities CanAddNewLayer = 0x1;
+      /** Flag to indicate that new features can be added to an existing layer */
+      CanAppendToExistingLayer       = 1 << 1,
 
-    /** Flag to indicate that new features can be added to an existing layer */
-    static const EditionCapabilities CanAppendToExistingLayer = 0x2;
+      /** Flag to indicate that new fields can be added to an existing layer. Imply CanAppendToExistingLayer */
+      CanAddNewFieldsToExistingLayer = 1 << 2,
 
-    /** Flag to indicate that new fields can be added to an existing layer. Imply CanAppendToExistingLayer */
-    static const EditionCapabilities CanAddNewFieldsToExistingLayer = 0x4;
+      /** Flag to indicate that an existing layer can be deleted */
+      CanDeleteLayer                 = 1 << 3
+    };
 
-    /** Flag to indicate that an existing layer can be deleted */
-    static const EditionCapabilities CanDeleteLayer = 0x8;
+    /** Combination of CanAddNewLayer, CanAppendToExistingLayer, CanAddNewFieldsToExistingLayer or CanDeleteLayer
+      * @note Added in QGIS 3.0 */
+    Q_DECLARE_FLAGS( EditionCapabilities, EditionCapability )
```

Comment:
Two more bits required - https://github.com/qgis/QGIS/blob/master/src/core/qgsvectordataprovider.h#L493

and https://github.com/qgis/QGIS/blob/master/python/core/qgsvectordataprovider.sip#L400